src/Workspaces/Remote/ServiceHub/Shared/RoslynJsonConverter.RoslynOnly.cs

```diff
@@ -344,13 +345,13 @@ protected override void WriteValue(JsonWriter writer, AddImportFixData source, J
                 writer.WriteValue((int)source.Kind);
 
                 writer.WritePropertyName(nameof(AddImportFixData.TextChanges));
-                serializer.Serialize(writer, source.TextChanges);
+                serializer.Serialize(writer, source.TextChanges ?? SpecializedCollections.EmptyList<TextChange>());
 
                 writer.WritePropertyName(nameof(AddImportFixData.Title));
                 writer.WriteValue(source.Title);
 
                 writer.WritePropertyName(nameof(AddImportFixData.Tags));
-                serializer.Serialize(writer, source.Tags.NullToEmpty());
+                serializer.Serialize(writer, source.Tags ?? SpecializedCollections.EmptyList<string>());
```

Comment:
i don't think this is right.  NullToEmpty is an extension method on ImmutableArray.  That means source.Tags is an immutable array.  And immutable arrays are value types and will never be null for ??.